salt/_modules/metalk8s.py

```diff
@@ -214,26 +239,8 @@ def get_archives(archives=None):
 
     res = {}
     for archive in archives:
-        if os.path.isdir(archive):
-            iso = None
-            info = archive_info_from_tree(archive)
-            path = archive
-            version = info['version']
-        elif os.path.isfile(archive):
-            iso = archive
-            info = archive_info_from_iso(archive)
-            version = info['version']
-            path = '/srv/scality/metalk8s-{0}'.format(version)
-        else:
-            log.warning(
-                'Skip, invalid archive path %s, should be an iso or a '
-                'directory.',
-                archive
-            )
-            continue
-
-        info.update({'path': path, 'iso': iso})
-        env_name = 'metalk8s-{0}'.format(version)
+        info = archive_info_from_product_txt(archive)
+        env_name = 'metalk8s-{0}'.format(info['version'])
 
         # Warn if we have 2 archives with the same version
         if env_name in res:
```

Comment:
I think we may want to raise in this case as well (maybe in another PR/ticket)

Reply:
Yes, I was wondering if we wanted to do that, I can handle it here.